### ISSUE TITLE
[Request] Add more sources for Windows

### ISSUE THREAD

**Issue by PlayingKarrde** [mmatyas/pegasus-frontend#671]
Like Steam and GOG.com, it would be great if there was support for:

- Uplay
- Epic
- Origin
- Twitch

**Comment by mmatyas** [comment]
Hm did Twitch start selling games too?

**Comment by PlayingKarrde** [comment]
Selling I couldn't say but they give away games a lot and have a launcher. Honestly this one would be much lower in importance, just added it for posterity really.

**Comment by PlayingKarrde** [comment]
Added Xbox as well as actually that would really be the most useful.

**Comment by sofalzone** [comment]
Do you intend to add additional platforms (i.e. Epic, Amazon)? I notice Playnite can bring in a lot of sources, but they don't show up in Pegasus through Playnite (i.e. Epic games show up in the Playnite launcher, but not under the Playnite source in Pegasus). I only see XBox/Windows games imported under the Playnite Pegasus source.

**Comment by mmatyas** [comment]
Yes, I plan to add support in the future, but contributions are also welcome, if someone is more familiar with these platforms.

At the moment only Steam games are imported from Playnite's own third-party support, but more could come in the future.

**Comment by blee264** [comment]
Not trying to necro a 4mo old thread but I was reading through this trying to think of a quick way to import EGL and Xbox/Microsoft games. Since Launchbox  already works to pull all the media for an epic or Xbox collection, wouldn't it be easier to implement through launchbox? The only thing that needs to be fixed is how Pegasus handles the launch commands from launchbox. I'm sure that's easier said than done but it seems much less involved than native support like Playnite. You can even alter the launch command in launchbox to suit Pegasus.

**Comment by HVR88** [comment]
Does it already work when importing games from Launchbox? While I use LB to populate Pegasus in Windows, I don't have any of the above-mentioned libraries to test this.

But... while it can be used as a quick stop-gap for anyone who wants to add those games now, it's important to keep Pegasus free of dependencies on other launchers. So, stop-gap use aside, it's not a replacement for @PlayingKarrde's request.

**Comment by blee264** [comment]
It pulls media and metadata perfectly. It just doesn't handle the launch command correctly so Pegasus crashes.

**Comment by blee264** [comment]
... it's important to keep Pegasus free of dependencies on other launchers...

What do you mean by that? I'm not disagreeing, I just don't think I understand what you're saying.

**Comment by HVR88** [comment]
LB is a third party commercial competitor to Pegasus. Requiring Launchbox for a feature like supporting other third-party stores makes, by definition, LB a "dependency." If Pegasus is to support additional stores/sources, they need to be baked into Pegasus itself.

Yes, Pegasus should ALSO work with these sources through Launchbox as Steam already does, but it can't be made to rely on Launchbox being installed as the only alternative to native support. Not everyone has LB nor wants LB.
